src/endpoints/mod_feedback.rs

```diff
@@ -0,0 +1,158 @@
+    data: web::Data<AppData>,
+    path: web::Path<GetModFeedbackPath>,
+    payload: web::Json<PostModFeedbackPayload>,
+    auth: Auth,
+) -> Result<impl Responder, ApiError> {
+    let dev = auth.developer()?;
+    let mut pool = data.db.acquire().await.or(Err(ApiError::DbAcquireError))?;
+    let mut transaction = pool.begin().await.or(Err(ApiError::TransactionError))?;
+
+    let access = Developer::has_access_to_mod(dev.id, &path.id, &mut transaction).await?;
+
+    if !access && !dev.verified && !dev.admin {
+        return Err(ApiError::Forbidden);
+    }
+
+    if access && payload.feedback_type != FeedbackTypeEnum::Note {
+        return Err(ApiError::Forbidden);
+    }
+
+    if !access && payload.feedback_type == FeedbackTypeEnum::Note {
+        return Err(ApiError::BadRequest("Only mod owners can leave notes".to_string()));
+    }
+
+    let mod_version = {
+        if path.version == "latest" {
+            ModVersion::get_latest_for_mod(&path.id, None, vec![], None, vec![ModVersionStatusEnum::Accepted, ModVersionStatusEnum::Pending, ModVersionStatusEnum::Rejected, ModVersionStatusEnum::Unlisted], &mut transaction).await?
+        } else {
+            ModVersion::get_one(path.id.strip_prefix('v').unwrap_or(&path.id), &path.version, false, false, &mut transaction).await?
+        }
+    };
+
+    let result = ModFeedback::set(&mod_version, dev.id, payload.feedback_type.clone(), &payload.feedback, false, &mut transaction).await;
+
+    if result.is_err() {
+        transaction
+            .rollback()
+            .await
+            .or(Err(ApiError::TransactionError))?;
+        return Err(result.err().unwrap());
+    }
+
+    transaction
+        .commit()
+        .await
+        .or(Err(ApiError::TransactionError))?;
+
+    Ok(HttpResponse::NoContent())
+}
+
+#[delete("/v1/mods/{id}/versions/{version}/feedback")]
+pub async fn delete_mod_feedback(
+    data: web::Data<AppData>,
+    path: web::Path<GetModFeedbackPath>,
+    auth: Auth,
+) -> Result<impl Responder, ApiError> {
+    let dev = auth.developer()?;
+    let mut pool = data.db.acquire().await.or(Err(ApiError::DbAcquireError))?;
+    let mut transaction = pool.begin().await.or(Err(ApiError::TransactionError))?;
+
+    let access = Developer::has_access_to_mod(dev.id, &path.id, &mut transaction).await?;
+
+    if !access && !dev.verified && !dev.admin {
+        return Err(ApiError::Forbidden);
+    }
+
+    let mod_version = {
+        if path.version == "latest" {
+            ModVersion::get_latest_for_mod(&path.id, None, vec![], None, vec![ModVersionStatusEnum::Accepted, ModVersionStatusEnum::Pending, ModVersionStatusEnum::Rejected, ModVersionStatusEnum::Unlisted], &mut transaction).await?
+        } else {
+            ModVersion::get_one(path.id.strip_prefix('v').unwrap_or(&path.id), &path.version, false, false, &mut transaction).await?
+        }
+    };
+
+    let result = ModFeedback::remove(&mod_version, dev.id, &mut transaction).await;
```

Comment:
are there no rules on deleting feedback? i think only admin and the dev who wrote the feedback should be allowed to delete their comments

Reply:
you can only delete your own feedback

Reply:
ah yeah, i didn't realize, imo i think this should be based on the feedback id anyways (you'd have to return the id in the list endpoint, but that's fine)

mostly so that admins could delete bad pieces of feedback if necessary